pkg/workceptor/kubernetes.go

```diff
@@ -203,7 +206,15 @@ func (kw *kubeUnit) createPod(env map[string]string) error {
 			return kw.clientset.CoreV1().Pods(ked.KubeNamespace).Watch(kw.ctx, options)
 		},
 	}
-	ev, err := watch2.UntilWithSync(kw.ctx, lw, &corev1.Pod{}, nil, podRunningAndReady)
+
+	timeout, err := strconv.Atoi(ked.PodPendingTimeout)
+	ctxWithTimeout := kw.ctx
+	if err != nil {
+		logger.Debug("Error parsing timeout, using infinte timeout")
```

Comment:
if there is an error parsing the timeout param we should error and return

Reply:
we should probably parse the duration earlier, like in SetFromParams -- the earlier in the code we can return an error the better. We can parse into a time.Duration object and put that on the kubeUnit struct directly, not the string representation.

Reply:
I have made the required changes. The error gets returned early now.